kube/src/api/subresource.rs

```diff
@@ -208,3 +210,134 @@ where
         Ok(self.client.request_text_stream(req).await?)
     }
 }
+
+// ----------------------------------------------------------------------------
+// Attach subresource
+// ----------------------------------------------------------------------------
+/// Params for attaching
+#[derive(Default)]
+pub struct AttachParams<R, W>
+where
+    R: AsyncRead + Unpin,
+    W: AsyncWrite + Unpin,
+{
+    /// The container for which to attach. Defaults to only container if there is one container in the pod.
+    pub container: Option<String>,
+    /// If provided, the standard out stream of the pod will be redirected to it.
+    pub stdout: Option<W>,
+    /// If provided, the standard error stream of the pod will be redirected to it.
+    pub stderr: Option<W>,
+    /// If provided, the contents will be redirected to the standard input stream of the pod.
+    pub stdin: Option<R>,
+    /// Allocate a terminal for this attach call; defaults to `false`.
+    pub tty: bool,
+}
```

Comment:
This looks very nice. Do you have some sources of what this is based on?

Reply:
For some reason, I couldn't (still can't) find it on https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#pod-v1-core, but I got the query parameters from OpenShift's docs.

https://docs.openshift.com/enterprise/3.0/rest_api/kubernetes_v1.html#connect-post-requests-to-attach-of-pod

Reply:
Ah nice. There's also https://docs.rs/k8s-openapi/0.10.0/k8s_openapi/api/core/v1/struct.ConnectGetNamespacedPodAttachOptional.html which seems similar

Reply:
It turns out it only accepts `GET`. I don't know why it has `POST` documented.